File: kotlin/darts/src/main/kotlin/Darts.kt
import kotlin.math.sqrt

object Darts {

    fun score(x: Number, y: Number): Int {
        val radius = sqrt(x.toDouble() * x.toDouble() + y.toDouble() * y.toDouble())

        return when {
            radius <= 1 -> 10
            radius <= 5 -> 5
            radius <= 10 -> 1
            else -> 0
        }
    }
}

File: cpp/darts/darts.cpp
#include "darts.h"
#include "math.h"

namespace darts
{

    int score(float x, float y)
    {
        auto fromO = sqrt(pow(x, 2) + pow(y, 2));

        if (fromO > 10)
        {
            return 0;
        }

        if (fromO > 5)
        {
            return 1;
        }

        if (fromO > 1)
        {
            return 5;
        }

        return 10;
    }

} // namespace darts
